_posts/2025-02-17-using-langchain4j-to-analyze-pdf-documents.adoc

```diff
@@ -0,0 +1,183 @@
+----
+
+Next we need to install the LangChain4j dependencies:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkiverse.langchain4j</groupId>
+    <artifactId>quarkus-langchain4j-core</artifactId>
+    <version>0.24.0</version>
+</dependency>
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j-google-ai-gemini</artifactId>
+    <version>1.0.0-beta1</version>
+</dependency>
+----
+
+Next we need to wire up the Gemini LLM to the application (using your Google AI Gemini API key).
+
+[source,java]
+----
+@ApplicationScoped
+public class GoogleGeminiConfig {
+
+    @Produces
+    @ApplicationScoped
+    ChatLanguageModel model() {
+        return GoogleAiGeminiChatModel.builder()
+                .apiKey(System.getenv("GOOGLE_AI_GEMINI_API_KEY"))
+                .modelName("gemini-2.0-flash")
+                .build();
+    }
+}
+----
+
+Now we need to model the data structure for the lease information that we want the LLM to extract from any lease document.  You can customize these fields based on the information you need from the PDF documents but in our use case below we are extracting the following information:
+
+[source,java]
+----
+public record LeaseReport(
+    LocalDate agreementDate,
+    LocalDate termStartDate,
+    LocalDate termEndDate,
+    LocalDate developmentTermEndDate,
+    String landlordName,
+    String tenantName,
+    String taxParcelId,
+    BigDecimal acres,
+    Boolean exclusiveRights) {
+}
+----
+
+Lastly, we need to create a `LeaseAnalyzerResource` class that will use the LLM to extract the lease information from the PDF document.
+
+[source,java]
+----
+@PUT
+@Consumes(MediaType.MULTIPART_FORM_DATA)
+@Produces(MediaType.TEXT_PLAIN)
+public String upload(@RestForm("file") FileUpload fileUploadRequest) {
+    final String fileName = fileUploadRequest.fileName();
+    log.infof("Uploading file: %s", fileName);
+
+    try {
+        // Convert input stream to byte array for processing
+        byte[] fileBytes = Files.readAllBytes(fileUploadRequest.filePath());
+
+        // Encode PDF content to base64 for transmission
+        String documentEncoded = Base64.getEncoder().encodeToString(fileBytes);
+
+        // Create user message with PDF content for analysis
+        UserMessage userMessage = UserMessage.from(
+                TextContent.from("Analyze the given document"),
+                PdfFileContent.from(documentEncoded, "application/pdf"));
+
+        // Build chat request with JSON response format
+        ChatRequest chatRequest = ChatRequest.builder()
+                .messages(userMessage)
+                .parameters(ChatRequestParameters.builder()
+                        .responseFormat(responseFormatFrom(LeaseReport.class))
+                        .build())
+                .build();
+
+        log.info("Google Gemini analyzing....");
+        long startTime = System.currentTimeMillis();
+        ChatResponse chatResponse = model.chat(chatRequest);
```

Comment:
I assume `model` is obtained via CDI, right? I think it makes sense to show that code too.

Reply:
Added @Inject in the code example